File: biosilico-creatures/src/main/resources/creatures/articlesAroundAndScripts/scripts/test_creatures_parser.py
import unittest
import os
from creatures1genomeAnalysis import parse_genome_file, export_to_json

class TestCreaturesParser(unittest.TestCase):
    def setUp(self):
        self.test_dir = "test_genomes"
        self.valid_files = ["valid_genome1.gen", "valid_genome2.gen", "minimal_genome.gen"]
        self.corrupted_files = ["corrupted_header.gen", "corrupted_data.gen"]
        self.empty_file = "empty.gen"

    def test_valid_genomes(self):
        for filename in self.valid_files:
            file_path = os.path.join(self.test_dir, filename)
            genes = parse_genome_file(file_path)
            self.assertIsInstance(genes, list)
            self.assertGreater(len(genes), 0)
            for gene in genes:
                self.assertIsNotNone(gene.type)
                self.assertIsNotNone(gene.subtype)

    def test_empty_file(self):
        file_path = os.path.join(self.test_dir, self.empty_file)
        genes = parse_genome_file(file_path)
        self.assertEqual(len(genes), 0)

    def test_corrupted_files(self):
        for filename in self.corrupted_files:
            file_path = os.path.join(self.test_dir, filename)
            genes = parse_genome_file(file_path)
            self.assertIsInstance(genes, list)  # Doit retourner une liste vide ou partielle
            print(f"Parsed {len(genes)} genes from {filename} (corrupted)")

    def test_export_json(self):
        file_path = os.path.join(self.test_dir, self.valid_files[0])
        genes = parse_genome_file(file_path)
        export_to_json(genes, "test_output.json")
        self.assertTrue(os.path.exists("test_output.json"))

if __name__ == "__main__":
    unittest.main()

File: biosilico-creatures/src/main/resources/creatures/articlesAroundAndScripts/scripts/test_creatures_parser.pl
#!/usr/bin/perl
use strict;
use warnings;
use Test::More;
use File::Basename;

require_ok('./creatures1genomeAnalysis.pl');

my $test_dir = "test_genomes_scripts";
my @valid_files = ("valid_genome1.gen", "valid_genome2.gen", "minimal_genome.gen");
my @corrupted_files = ("corrupted_header.gen", "corrupted_data.gen");
my $empty_file = "empty.gen";

sub test_valid_genomes {
    for my $filename (@valid_files) {
        my $file_path = "$test_dir/$filename";
        my $genes = parse_genome_file($file_path);
        is(ref($genes), 'ARRAY', "$filename: returns an array");
        ok(@$genes > 0, "$filename: has genes");
        for my $gene (@$genes) {
            ok(defined $gene->{type}, "$filename: gene has type");
            ok(defined $gene->{subtype}, "$filename: gene has subtype");
        }
    }
}

sub test_empty_file {
    my $file_path = "$test_dir/$empty_file";
    my $genes = parse_genome_file($file_path);
    is(scalar(@$genes), 0, "$empty_file: no genes parsed");
}

sub test_corrupted_files {
    for my $filename (@corrupted_files) {
        my $file_path = "$test_dir/$filename";
        my $genes = parse_genome_file($file_path);
        is(ref($genes), 'ARRAY', "$filename: returns an array (may be empty)");
        diag(sprintf("%s: parsed %d genes (corrupted)", $filename, scalar(@$genes)));
    }
}

test_valid_genomes();
test_empty_file();
test_corrupted_files();

done_testing();
